src/main/java/org/ow2/proactive/connector/iaas/cloud/provider/jclouds/aws/AWSEC2JCloudsProvider.java

```diff
@@ -303,6 +312,28 @@ public void removeInstancePublicIp(Infrastructure infrastructure, String instanc
         }
     }
 
+    @Override
+    public void deleteKeyPair(Infrastructure infrastructure, String keyPairName, String region) {
+        KeyPairApi keyPairApi = getKeyPairApi(infrastructure);
+        keyPairApi.deleteKeyPairInRegion(region, keyPairName);
+        log.info("Removed the key pair [{}] in the region [{}]", keyPairName, region);
```

Comment:
I thought this logger doesn't support directly text formatting. Cf. L329.

Reply:
As this class is using Log4j2, so it support text formatting (and it's tested).